aten/src/ATen/core/thread_pool.cpp

```diff
@@ -3,23 +3,127 @@
 
 namespace c10 {
 
-ThreadPool::ThreadPool() { /* intended to leave empty */ }
+ThreadPool::ThreadPool(std::size_t pool_size, int numa_node_id)
+    : threads_(pool_size),
+      running_(true),
+      complete_(true),
+      available_(pool_size),
+      total_(pool_size),
+      numa_node_id_(numa_node_id) {
+  for (std::size_t i = 0; i < pool_size; ++i) {
+    threads_[i] = std::thread(std::bind(&ThreadPool::main_loop, this, i));
+  }
+}
+
+ThreadPool::~ThreadPool() {
+  // Set running flag to false then notify all threads.
+  {
+    std::unique_lock<std::mutex> lock(mutex_);
+    running_ = false;
+    condition_.notify_all();
+  }
+
+  try {
+    for (auto& t : threads_) {
+      t.join();
```

Comment:
should we try..catch t.join() to make sure we don't miss a thread to join before exiting from desctructor